tests/test_config.py

```diff
@@ -0,0 +1,60 @@
+from pytest_mock_resources.compat import mock
+
+from pytest_mock_resources.config import DockerContainerConfig, fallback, get_env_config
+
+_DOCKER_HOST = "host.docker.internal"
+
+
+class Test_get_env_config:
+    def test_it_is_missing_env_var(self):
+        with mock.patch("os.environ", {}):
+            value = get_env_config("postgres", "username")
+            assert value is None
+
+    def test_it_has_the_expected_env_var(self):
+        with mock.patch("os.environ", {"PMR_POSTGRES_USERNAME": "foobar"}):
+            value = get_env_config("postgres", "username")
+            assert value == "foobar"
+
+
+class FooConfig(DockerContainerConfig):
+    name = "foo"
+    _fields = {"image", "host", "port", "ci_port", "extra_config", "no_value_default"}
+    _fields_defaults = {"image": "test", "extra_config": "bar"}
+
+    @fallback
+    def extra_config(self):
+        raise NotImplementedError()
+
+    @fallback
```

Comment:
it would be ideal if the user did not have to import and levarage this - maybe this could be on some underscored method which ultimately calls this fn

Reply:
They dont, see the added docs. This is for **us** defining the interface to the config fixture.

a user would write
```
from pytest_mock_resources import FooConfig

@pytest.fixture
def pmr_foo_config():
   return FooConfig(image='foo:latest')
```

Reply:
ah, alright, makes sense